_posts/2020-01-12-updating-foreman-inventory-with-system-facts.md

```diff
@@ -0,0 +1,311 @@
+    </tr>
+    <tr>
+      <td>Default location</td>
+      <td>Puppet</td>
+      <td>When host is created based on either facts, we search for a fact representing the location. If there's no such fact or the host  is created based on the report, it will be assigned to the location specified in this setting.</td>
+    </tr>
+    <tr>
+      <td>Default organization</td>
+      <td>Puppet</td>
+      <td>Same as default location, but works for organization</td>
+    </tr>
+    <tr>
+      <td>Location fact</td>
+      <td>Puppet</td>
+      <td>The name of the fact, which contains the name of the location, in which this host should be placed. By default, there's no such  fact, but all configuration management systems allow to specify custom fact. Hence you can create a custom fact e.g. foreman_location and then the Host will be automatically assigned to it. </td>
+    </tr>
+    <tr>
+      <td>Organization fact</td>
+      <td>Puppet</td>
+      <td>Same as above but for organization</td>
+    </tr>
+    <tr>
+      <td>Update subnets from facts</td>
+      <td>Puppet</td>
+      <td>If an IP of existing interface is changed (or newly created), Foreman will try to match existing Subnet object based on this setting.</td>
+    </tr>
+    <tr>
+      <td>Exclude pattern for facts stored in foreman</td>
+      <td>Provisioning</td>
+      <td>Facts can contain a lot of data that may not be interesting. E.g. if you don't want to import all blockdevices from each host, all network interfaces etc, you can specify a pattern for ignoring. The default value contains commonly ignored patterns, e.g. "docker&ast;"</td>
+    </tr>
+    <tr>
+      <td>Ignore facts for domain</td>
+      <td>Provisioning</td>
+      <td>By default, Foreman will be creating and assigning Domain objects based on the host domain name. If you don't want this to be updated, you can disable this behavior using this setting.</td>
+    </tr>
+    <tr>
+      <td>Ignore facts for operating system</td>
+      <td>Provisioning</td>
+      <td>By default, Foreman will be creating and assigning Operating System objects based on OS facts. This is useful when OS is upgraded on the host, but it will also create a new OS object in Foreman, which is not configured fully in order to be provisioned. When you decide to reprovision the host, you may be surprised you need to provide additional configuration for the Operating System first</td>
+    </tr>
+    <tr>
+      <td>Ignore interfaces with matching identifier</td>
+      <td>Provisioning</td>
+      <td>This is similar to excluded pattern settings above, it's only limited to parsing network interfaces facts. It existed earlier but should probably be removed now.</td>
+    </tr>
```

Comment:
You can explain that creating a NIC is more expensive operation, this gives you the flexibility to store facts but not creating those NICs. It's still useful, I'd avoid commenting on what should be removed in the future. This is not the right place.